### PR TITLE
DIV-5824 DIV-5813 - sync missing prod changes into master

### PR DESCRIPTION
**Pull request by hosainnet**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
DIV-5824 DIV-5813

### Change description ###

Steps taken:
1) Created a branch out of master
2) Merged `prod` into master
3) Fixed (A LOT) of "conflicts" (mostly the same changes under a different commit
4) Reverted https://github.com/hmcts/div-ccd-definitions/commit/6496dc93634700276ec8f10faba19a5d6ba1813d (refusal stuff, not ready for prod)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 1]
```diff
@@ -12634,160 +12634,6 @@
     "UserRole": "caseworker-divorce-solicitor",
     "CRUD": "R"
   },
-  {
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: @qzhou-hmcts do we have to feature flag some code or tests as we're reverting these?

File: definitions/divorce/json/CaseEvent.json [hunk 2]
```diff
@@ -2097,7 +2097,7 @@
     "PreConditionState(s)": "AwaitingLegalAdvisorReferral",
     "PostConditionState": "*",
     "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/dn-about-to-be-granted",
-    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/dn-decision-made",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/clean-state",
```
Comment (hosainnet) [inline definitions/divorce/json/CaseEvent.json]: @tchow8 is this okay?

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 1 above)
Comment (qzhou-hmcts) [reply definitions/divorce/json/AuthorisationCaseField.json]: COS already has the feature flag

File: definitions/divorce/json/CaseEvent.json (hunk 2 above)
Comment (qzhou-hmcts) [reply definitions/divorce/json/CaseEvent.json]: this is fine, dn-decision-made will be release with refusal

**Pull request merged**